src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -418,7 +418,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         });
 
         if (this.exercise!.type === ExerciseType.MODELING || this.exercise!.type === ExerciseType.TEXT) {
-            return this.gradedStudentParticipation?.results?.find((result: Result) => !!result.completionDate) || undefined;
+            return;
```

Comment:
When you are already at it, you can specify the return type of this method in its signature. This way, errors like this get caught by the compiler.